core/app/models/spree/order_promotion.rb

```diff
@@ -2,5 +2,15 @@ module Spree
   class OrderPromotion < Spree::Base
     belongs_to :order, class_name: 'Spree::Order'
     belongs_to :promotion, class_name: 'Spree::Promotion'
+
+    delegate :name, :description, to: :promotion
+    delegate :currency, to: :order
+
+    extend Spree::DisplayMoney
+    money_methods :amount
+
+    def amount
+      order.all_adjustments.promotion.where(source: [promotion.actions]).sum(:amount)
```

Comment:
Are these square brackets needed here? I think we can write `where(source: promotion.actions)`.